src/pages/kyubinkim/Main/Main.js

```diff
@@ -0,0 +1,255 @@
+import React, { Component } from "react";
+import "../Main/Main.scss";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {
+  faSearch,
+  faUserCircle,
+  faBookmark,
+  faCog,
+  faHeart,
+  faComment,
+  faShare,
+} from "@fortawesome/free-solid-svg-icons";
+import { Link } from "react-router-dom";
+import ListComponent from "./ListComponent";
+
+class MainKyubin extends Component {
+  state = {
+    userComment: "",
+    userCommentList: [],
+  };
+
+  componentDidMount() {
+    fetch("http://localhost:3000/data/Data.json", {
+      method: "GET",
+    })
+      .then((res) => res.json())
+      .then((res) => {
+        this.setState({
+          userCommentList: res.data,
+        });
+      });
+  }
+
+  userComment = (e) => {
+    this.setState({ userComment: e.target.value });
+  };
+
+  handleCreate = (e) => {
+    e.preventDefault();
+    const { userCommentList, userComment } = this.state;
+    if (this.state.userComment !== "") {
+      this.setState({
+        userCommentList: [
+          ...userCommentList,
+          {
+            id: userCommentList.length + 1,
+            mention: userComment,
+            inLike: 0,
+            likePosition: false,
+          },
+        ],
+      });
+      this.setState({ userComment: "" });
+    }
+  };
```

Comment:
```suggestion
  handleCreate = (e) => {
    e.preventDefault();
    const { userCommentList, userComment } = this.state;
    if (userComment === "") return;
    
    const newUserCommentList = [
          ...userCommentList,
          {
            id: userCommentList.length + 1,
            mention: userComment,
            inLike: 0,
            likePosition: false,
          },
    ];
    
    this.setState({ 
      userCommentList: newUserCommentList,
      userComment: "" });
    }
  };
```

- 예외 처리를 먼저 해서 return 시킬 수 있습니다.
- 변수를 잘 활용하면 코드가 깔끔해질 수 있습니다.
- spread 연산자 활용 Good.
- state 객체 데이터를 한번에 변경할 수 있습니다.

Reply:
예외 처리에서 리턴에서 저게 가능한 이유가 ( 리턴을 만나면 함수가 종료되니까 ) 맞나요?

저 함수새로운 변수에 담고 setState 로 선언 ! 감사합니다